### PR TITLE
Bratseth/compare equals

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

File: container-search/src/main/java/com/yahoo/prelude/query/parser/SpecialTokens.java [hunk 1]
```diff
@@ -142,16 +142,23 @@ public String replace() {
...
+        @Override
         public int compareTo(SpecialToken other) {
-            if (this.token().length() < other.token().length()) {
-                return 1;
-            } else if (this.token().length() == other.token().length()) {
-                return 0;
-            } else {
-                return -1;
-            }
+            if (this.token().length() < other.token().length()) return 1;
```
Comment (baldersheim) [inline container-search/src/main/java/com/yahoo/prelude/query/parser/SpecialTokens.java]: longer token sorts before shorter ?
Reverse of normal alphabetical sorting ?

Comment (bratseth) [reply container-search/src/main/java/com/yahoo/prelude/query/parser/SpecialTokens.java]: Yes, this is odd but I didn't want to change anything here (except in Relevance).

**Review by baldersheim — APPROVED** [review]
Apart from the single comment this looks fine.

File: container-search/src/main/java/com/yahoo/prelude/query/parser/SpecialTokens.java (hunk 1 above)
Comment (bratseth) [reply container-search/src/main/java/com/yahoo/prelude/query/parser/SpecialTokens.java]: Btw, I think the longer tokens first thing is to prefer matching longer tokens.

**Pull request merged**